cluster-autoscaler/cloudprovider/azure/azure_scale_set.go

```diff
@@ -379,7 +379,18 @@ func buildGenericLabels(template compute.VirtualMachineScaleSet, nodeName string
 	result[kubeletapis.LabelOS] = buildInstanceOS(template)
 	result[kubeletapis.LabelInstanceType] = *template.Sku.Name
 	result[kubeletapis.LabelZoneRegion] = *template.Location
-	result[kubeletapis.LabelZoneFailureDomain] = "0"
+
+	if len(*template.Zones) > 0 {
+		failureDomains := make([]string, len(*template.Zones))
+		for k, v := range *template.Zones {
+			failureDomains[k] = *template.Location + "-" + v
+		}
+
+		result[kubeletapis.LabelZoneFailureDomain] = strings.Join(failureDomains[:], kubeletapis.LabelMultiZoneDelimiter)
+	} else {
+		result[kubeletapis.LabelZoneFailureDomain] = "0"
+	}
+
```

Comment:
@feiskyer can you review this part alone ?

Reply:
sure. added two comments.